### ISSUE TITLE
Share Place Name Search and Find

### ISSUE THREAD

**Issue by HerzScheisse** [vesta-webtrees-2-custom-modules/vesta_shared_places#172]
Hi, its me again with a kinda rare and maybe exotic error/problem ;-)

So i need to explain a bit, so you understand how i handle my places...

for records of any kind (BIRT, RESI, DEAT etc..) i try to keep the name of the place clean (only the town with hierarchy) f.e. `Dresden, Sachsen, Deutschland`
for the shared place I'll add the location/streetname with its todays valid name...
f.e. `Zwickauer Straße 14, Dresden, Sachsen, Deutschland`
for the ADDR i'll add the place/street name how it was called in that time when it was recorded
f.e. `An der Falkenbrücke 2a`

so in the above example the shared place is `Zwickauer Straße 14` and i can add to this place also other name variations... 
if i need to add this shared place to another INDI, even if i'll type in the shared place input field `An der Falkenbrücke` the shared place Zwickauer Straße is suggested... works as expected...
BUT the downside is, that i need to add for every name variation the gps coordinates in the geografical data... I try to avoid that...

My thinking is now, that i rename the shared place to `Zwickauer Straße 14 (An der Falkenbrücke 2a)` but this has a downside too, because if i type `Falkenbrücke` in the shared place input field, nothing get suggested, because the ajax? function works only if you start with the correct begin of the shared place...

**long story short**, is it possible that we can also filter in the shared place input field the names which dont exactly start with the text we input?

hope this makes any sense to you and you understood what i mean :)

![an der falkenbrücke](https://github.com/vesta-webtrees-2-custom-modules/vesta_shared_places/assets/146635896/2c48c130-6fee-4c08-8999-01d283e76ac7)
![an der falkenb](https://github.com/vesta-webtrees-2-custom-modules/vesta_shared_places/assets/146635896/9e99d9e3-9246-4aab-8e54-53362f676fd8)

P.S.: while adding the screenshots i noticed a untranslated string "No results found" maybe we can fix and add a translation too?! :-)

**Comment by ric2016** [comment]
> for records of any kind (BIRT, RESI, DEAT etc..) i try to keep the name of the place clean (only the town with hierarchy) 
> f.e. Dresden, Sachsen, Deutschland
> for the shared place I'll add the location/streetname with its todays valid name...
> f.e. Zwickauer Straße 14, Dresden, Sachsen, Deutschland

I wouldn't recommend to do it like this. Why no use the full hierarchy in both cases? Basically, the link to the shared place is intended to replace the PLAC name of the event, which is however still required due to the GEDCOM spec. (Actually, the explicit PLAC name is still useful if the shared place has multiple names, as described below)

But of course that's up to you.

----

In any case, if the shared place has different names at different times, that should be modelled explicitly, not like this:

> i rename the shared place to Zwickauer Straße 14 (An der Falkenbrücke 2a)

You can add multiple names to a shared place, and define date ranges for each name. The alternate names are then available via the search. The date ranges are actually optional, they are useful in case you want to fill in the proper PLAC name of an event automatically via the shared place.

```
1 NAME Breslau
2 DATE TO 18 MAY 1946
1 NAME Wrocław
2 DATE FROM 19 MAY 1946
```

----

Regarding the actual feature request: It may still be useful to adjust the search as suggested in general, I'll have to look into that.

**Comment by ric2016** [comment]
> P.S.: while adding the screenshots i noticed a untranslated string "No results found"

This is a general webtrees issue, reproducible on the demo server: Searching for sources, media objects etc. also returns "No results found" untranslated.

**Comment by HerzScheisse** [comment]
Thanks for your reply and insights and in genrall for this module... i really love and apprecciate this...

I've been thinking and playing with this for almost a year... With a lot of attempts how to handle it best, my above described way works best for me...

>Why no use the full hierarchy in both cases?

First problem is, if i use in both places the full hierarchy, then in reports where i want to display only the first part of the place, i only get the f.e. a streetadrdress...
I always add the streetaddress in the ADDR field... so while viewing the tree and INDI profiles, i have a direkt view where the person lived...
Second problem is, I want to group all places/strrets/homes where ancestors lived... even if the place name changed over time...

> You can add multiple names to a shared place, and define date ranges for each name.

yeah i found this by playing with the places and shared places... but didnt know that we can add date ranges... this was new to me...
if I add each variation of a name to the shared place... i also need to add the location coord. twice at least... 

> Regarding the actual feature request: It may still be useful to adjust the search as suggested in general, I'll have to look into that.

Thanks! it would be great if this can be added at some point... 
my wanted functionallity generally works in webtrees, f.e. in the add source input field...

**Comment by HerzScheisse** [comment]
> > P.S.: while adding the screenshots i noticed a untranslated string "No results found"
> 
> This is a general webtrees issue, reproducible on the demo server: Searching for sources, media objects etc. also returns "No results found" untranslated.

so i guess i have to open a issue at webtrees directly, rigtht?

**Comment by ric2016** [comment]
> if I add each variation of a name to the shared place... i also need to add the location coord. twice at least...

Why twice? You just have to add the coordinates to the shared place once, regardless of the number of names.

**Comment by HerzScheisse** [comment]
If i add a second name to a exsiting shared place (the existing one has the location data set) i need to add for the new added name variation also the location data in control panel > geografical data...

or is there a other way to avoid this? if you like i can provide you a login to my side where you can have a look at my settings..

**Comment by ric2016** [comment]
> i need to add for the new added name variation also the location data in control panel > geografical data

If you use shared places, you no longer have to use webtrees geographical data at all. There is a data fix that moves the coordinates to shared places.

**Comment by HerzScheisse** [comment]
> If you use shared places, you no longer have to use webtrees geographical data at all.
hmmm... interessting... its the first time that i read about this...
it could be that i'm doing it "wrong" but with the georafical data in control panel it was very easy to submit the geo coords 

and even if i'm in the shared place list the edit icon (if geo coords are missing) links to the geografical data in CP
![shared place add loc](https://github.com/vesta-webtrees-2-custom-modules/vesta_shared_places/assets/146635896/508932c9-eb4c-41f8-9191-dcbc941fd1e5)

> There is a data fix that moves the coordinates to shared places.
yes, i've been using this the whole time.... maybe also wrong :-D

my workflow was like this:
- adding data the the tree
- from time to time check geografical data and fix the missing geo coords
- run the vesta data fix (option 2)

**Comment by ric2016** [comment]
> and even if i'm in the shared place list the edit icon (if geo coords are missing) links to the geografical data in CP

That link is inherited from the original impl (it's also useful if the place is no shared place).

> my workflow was like this:

The data fix wasn't intended to be part of a regular workflow, it was designed as a tool for one-time migration. But of course it's possible to use it like you describe.

But if there was an option to search for shared places with missing coordinates, the process could be simplified by adding the coordinates directly to the shared place.

**Comment by ric2016** [comment]
Actual issue

> is it possible that we can also filter in the shared place input field the names which dont exactly start with the text we input?

is fixed in latest commit. I think it only accidentally matched the start of the name only.

**Comment by HerzScheisse** [comment]
> could be simplified by adding the coordinates directly to the shared place.

yeah... but then i have to add and pick them manually, right? thats why i liked the usage of the geografical places in CP...
I think i need to rethink again how to handle this... 

> is fixed in latest commit. I think it only accidentally matched the start of the name only.

Thanks! I think i will wait some days before upgrading wt and your modlues... but thanks for fixing :)

**Comment by HerzScheisse** [comment]
for the record, the issue  is now fixed with https://github.com/fisharebest/webtrees/pull/4949
